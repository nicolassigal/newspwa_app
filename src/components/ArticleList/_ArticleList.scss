.article-list {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(300px,1fr));
    grid-gap: 30px;
    margin: 0;
    padding: 0;
    top: 30px;
    padding-bottom: 30px;
    position: relative;
}