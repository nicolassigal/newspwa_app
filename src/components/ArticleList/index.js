import React from "react";
import Article from '../../components/Article';

const ArticleList = ({ articles }) => {
  return (
   <ul className="article-list">
      {articles.map((article, i) => (
          <Article
            key={i}
            title={article.title}
            image={article.urlToImage}
            url={article.url}
            description={article.description} />
        ))}
    </ul>
  );
};

export default ArticleList;
