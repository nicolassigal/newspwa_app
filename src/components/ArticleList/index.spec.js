import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ArticleList from './index';
import { shallow } from 'enzyme';
import "intersection-observer";

describe('ArticleList component test', () => {
  it('ArticleList component should render correctly', () => {
    const props = {
      articles: []  
    }


   const wrapper = shallow(<ArticleList {...props}/>)
   expect(wrapper).toMatchSnapshot();

  });

  it('ArticleList component should render with list', () => {
    const props = {
        articles: [
            {
                title: 'title_1',
                description: 'desc_1',
                url: 'url_1',
                image:'image_1',
                key: '1'
            },
            {
                title: 'title_2',
                description: 'desc_2',
                url: 'url_2',
                image:'image_2',
                key: '2'
            }
        ]  
    }


   const wrapper = shallow(<ArticleList {...props}/>)
   expect(wrapper).toMatchSnapshot();

   const Articles = wrapper.find(".article-list").getElement();
   expect(Articles.props.children).toHaveLength(2);
  });
});