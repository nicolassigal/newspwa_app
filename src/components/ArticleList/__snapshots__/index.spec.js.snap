// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component test ArticleList component should render correctly 1`] = `ShallowWrapper {}`;

exports[`ArticleList component test ArticleList component should render with list 1`] = `ShallowWrapper {}`;
